gcc/config/riscv/rvp_intrinsic.h

```diff
@@ -0,0 +1,43 @@
+/* Builtin definitions for P extension
+   Copyright (C) 2021 Free Software Foundation, Inc.
+
+This file is part of GCC.
+
+GCC is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3, or (at your option)
+any later version.
+
+GCC is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with GCC; see the file COPYING3.  If not see
+<http://www.gnu.org/licenses/>.  */
+
+#ifndef _RISCV_RVP_INTRINSIC_H
+#define _RISCV_RVP_INTRINSIC_H
+
+typedef signed char int8x4_t __attribute ((vector_size(4)));
+typedef signed char int8x8_t __attribute ((vector_size(8)));
+typedef short int16x2_t __attribute ((vector_size(4)));
+typedef short int16x4_t __attribute__((vector_size (8)));
+typedef short int16x8_t __attribute__((vector_size (16)));
+typedef int int32x2_t __attribute__((vector_size(8)));
+typedef int int32x4_t __attribute__((vector_size(16)));
+typedef unsigned char uint8x4_t __attribute__ ((vector_size (4)));
+typedef unsigned char uint8x8_t __attribute__ ((vector_size (8)));
+typedef unsigned short uint16x2_t __attribute__ ((vector_size (4)));
+typedef unsigned short uint16x4_t __attribute__((vector_size (8)));
+typedef unsigned short uint16x8_t __attribute__((vector_size (16)));
+typedef unsigned int uint32x2_t __attribute__((vector_size(8)));
+typedef unsigned int uint32x4_t __attribute__((vector_size(16)));
+
+#define __riscv__kabsw(a) \
```

Comment:
it only helps intrinsic header more readable, e.g. 
```c
#if defined(__riscv_zpsf) && __riscv_xlen == 64
#define __rv__v_smar64 __riscv__v_smar64
#define __rv__v_smsr64 __riscv__v_smsr64
...
```
instead of several lines for one intrinsic. I can remove them if it is not appropriate.